tests/test_fs.py

```diff
@@ -770,46 +794,56 @@ def copydir_bad(tmppath, parallel):
 
     # Destination exists
     pytest.raises(
-        DestinationExistsError, fs.copydir, src_folder_exists, dst_exists,
+        DestinationExists, fs.copydir, src_folder_exists, dst_exists,
         parallel=parallel)
     pytest.raises(
-        DestinationExistsError, fs.copydir, src_folder_exists,
+        DestinationExists, fs.copydir, src_folder_exists,
         src_folder_exists, parallel=parallel)
     pytest.raises(
-        DestinationExistsError, fs.copydir, src_folder_exists,
+        DestinationExists, fs.copydir, src_folder_exists,
         dst_folder_exists, parallel=parallel)
 
     # Cannot move file
     pytest.raises(
-        ResourceInvalidError, fs.copydir, src_exists, dst_new,
+        ResourceInvalid, fs.copydir, src_exists, dst_new,
         parallel=parallel)
     pytest.raises(
-        ResourceInvalidError, fs.copydir, src_exists, dst_folder_new,
+        ResourceInvalid, fs.copydir, src_exists, dst_folder_new,
         parallel=parallel)
 
     # Source doesn't exists
     pytest.raises(
-        ResourceNotFoundError, fs.copydir, src_new, dst_exists,
+        ResourceNotFound, fs.copydir, src_new, dst_exists,
         parallel=parallel)
     pytest.raises(
-        ResourceNotFoundError, fs.copydir, src_new, dst_new,
+        ResourceNotFound, fs.copydir, src_new, dst_new,
         parallel=parallel)
     pytest.raises(
-        ResourceNotFoundError, fs.copydir, src_new, dst_folder_exists,
+        ResourceNotFound, fs.copydir, src_new, dst_folder_exists,
         parallel=parallel)
     pytest.raises(
-        ResourceNotFoundError, fs.copydir, src_new, dst_folder_new,
+        ResourceNotFound, fs.copydir, src_new, dst_folder_new,
         parallel=parallel)
 
     pytest.raises(
-        ResourceNotFoundError, fs.copydir, src_folder_new, dst_exists,
+        ResourceNotFound, fs.copydir, src_folder_new, dst_exists,
         parallel=parallel)
     pytest.raises(
-        ResourceNotFoundError, fs.copydir, src_folder_new, dst_new,
+        ResourceNotFound, fs.copydir, src_folder_new, dst_new,
         parallel=parallel)
     pytest.raises(
-        ResourceNotFoundError, fs.copydir, src_folder_new, dst_folder_exists,
+        ResourceNotFound, fs.copydir, src_folder_new, dst_folder_exists,
         parallel=parallel)
     pytest.raises(
-        ResourceNotFoundError, fs.copydir, src_folder_new, dst_folder_new,
+        ResourceNotFound, fs.copydir, src_folder_new, dst_folder_new,
         parallel=parallel)
+
+
+def test_openbin():
```

Comment:
Question: Are these new methods in the PyFilesystem API?

Reply:
Yes, see [here](https://docs.pyfilesystem.org/en/latest/implementers.html#essential-methods). For the time being, I didn't want to spend time to implement them.